src/panels/lovelace/cards/hui-alarm-panel-card.ts

```diff
@@ -217,26 +220,22 @@ class HuiAlarmPanelCard extends LitElement implements LovelaceCard {
   }
 
   private _handlePadClick(e: MouseEvent): void {
-    const input = this.shadowRoot!.querySelector(
-      "#alarmCode"
-    ) as PaperInputElement;
     const val = (e.currentTarget! as any).value;
-    input.value = val === "clear" ? "" : input.value + val;
+    this._input!.value = val === "clear" ? "" : this._input!.value + val;
   }
 
   private _handleActionClick(e: MouseEvent): void {
-    const input = this.shadowRoot!.querySelector(
-      "#alarmCode"
-    ) as PaperInputElement;
     const code =
-      input && input.value && input.value.length > 0 ? input.value : "";
+      this._input && this._input!.value && this._input!.value.length > 0
```

Comment:
Since you use `this._input` a lot, and are running `querySelector` every time in the background, it is better to store it in a local variable.
`const input = this._input!`

Reply:
Good point.